ASoC: pcm: send DAPM_STREAM_STOP event in dpcm_fe_dai_shutdown

ranj063 opened the pull request:
A recent change removed the call to send the DAPM_STREAM_STOP
event in dpcm_fe_dai_shutdown. But this causes a regression
when a PCM prepare is not paired with a hw_free. So, add
the DAPM_STREAM_STOP event back to dpcm_fe_dai_shutdown()
to fix this.

Fixes: a27b421f1d04 ('ASoC: pcm: call snd_soc_dapm_stream_stop() in soc_pcm_hw_clean')

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

plbossart commented:
@ranj063 can you add Reported-by: with Hans' email? We need to give him credit for his work.

bardliao commented:
But then snd_soc_dapm_stream_event SND_SOC_DAPM_STREAM_STOP will be triggered on both 
```
soc_pcm_hw_free
    -> soc_pcm_hw_clean
        -> snd_soc_dapm_stream_stop
            -> snd_soc_dapm_stream_event
```
```
dpcm_fe_dai_close
    -> dpcm_fe_dai_shutdown
        -> dpcm_dapm_stream_event;
```
Maybe we should post the question to mailing list?

ranj063 commented:
> But then snd_soc_dapm_stream_event SND_SOC_DAPM_STREAM_STOP will be triggered on both
> 
> ```
> soc_pcm_hw_free
>     -> soc_pcm_hw_clean
>         -> snd_soc_dapm_stream_stop
>             -> snd_soc_dapm_stream_event
> ```
> 
> ```
> dpcm_fe_dai_close
>     -> dpcm_fe_dai_shutdown
>         -> dpcm_dapm_stream_event;
> ```
> 
> Maybe we should post the question to mailing list?

@bardliao yes, but the second stop event doesn't.seem to do any harm.

plbossart commented:
maybe we should document the two sequences in the commit message to make sure this is captured and known by future generations.

ranj063 commented:
> maybe we should document the two sequences in the commit message to make sure this is captured and known by future generations.

done @plbossart . I added the sequences and added the reported-by and sent the patch upstream

kv2019i commented:
Merged upstrream as bb9dd3ce6177e1f8cf01b0d45e6bd9b93f656bd0 https://lore.kernel.org/r/20201202193343.912942-1-ranjani.sridharan@linux.intel.com 

Closing.
